zulip_bots/zulip_bots/bots/calendarbot/calendarbot.py

```diff
@@ -1,154 +1,250 @@
 import logging
-import os
+from typing import Any, Dict, List, Optional
 from dataclasses import dataclass
-from datetime import date, datetime, time, timedelta
-from tempfile import TemporaryFile
-from time import sleep
+
 from re import search
+from datetime import datetime, timedelta
+from tempfile import TemporaryFile
 
-from zulip_bots.bots.calendarbot.googlecalendar import GcalMeeting as GcalMeeting ,authenticate_google,AuthenticationError
 from ics import Attendee, Calendar, Event
 from zulip_bots.lib import BotHandler
+from zulip_bots.bots.calendarbot.googlecalendar import GcalMeeting as GcalMeeting,authenticate_google, AuthenticationError
+
+
+logging.basicConfig(
+    filename='zulip_bots/zulip_bots/bots/calendarbot/calendarbot.log',
+    encoding='utf-8',
+    level=logging.DEBUG
+)
 
-logging.basicConfig(filename='zulip_bots/zulip_bots/bots/calendarbot/calendarbot.log', encoding='utf-8', level=logging.DEBUG)
 
 #TODO: Take this from environment variables in the future
-BOT_REGEX = '.+(bot@).+(zulipchat.com)'
+BOT_REGEX = "(.*\*\*.*|.*bot.*)"
+TIME_FORMAT = "^(<time).*(>)$"
+DEFAULT_EVENT_DURATION = 30
+VIRTUAL_RC = "https://recurse.rctogether.com/"
 
-#create calendar_object
 
 class CalendarBotHandler(object):
     """
     This bot fucking loves meetings. 
     """
-    def init(self):
-        self.message = {}
-        self.bot_handler
-        
 
     @dataclass
     class MeetingDetails:
-        sender_id: int
-        invitees: list
+        name: str
+        location: str
+        description: str
         meeting_start: datetime
         meeting_end: datetime
-        name: str
-        summary: str
         length_minutes: int
+        sender_email: str
+        invitees: list
+
+
+    def initialize(self, bot_handler: BotHandler) -> None:
+        storage = bot_handler.storage
 
+        # Add key values in the bot_handler storage
+        if not storage.contains("confirmation"):
+            storage.put("confirmation", None)
+        if not storage.contains("starttime"):
+            storage.put("starttime", None)
+        if not storage.contains("duration"):
+            storage.put("duration", None)
 
-    def usage(self):
-        return """Creates calendar meeting for all participants on chat."""
 
-    def handle_message(self, message:dict, bot_handler):
-        self.message = message
-        self.bot_handler = bot_handler
-        message_content = message['content'].lower()
+    def usage(self) -> str:
+        return (
+            "Calendar Bot creates a calendar meeting for all participants on chat. "
+            "Enter a Zulip global time `<time` and then a duration in minutes (e.g. 60 for 1 hour). "
+            "Default event duration is 30 mins and default email addresses are from user accounts. "
```

Comment:
For the "30 mins" lets use the DEFAULT_DURATION variable instead of hardcoding it.